### PR TITLE
Generate code coverage with Coveralls

### PR DESCRIPTION
**Pull request by danvk**
I'll add a coverage badge as a follow-up.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 machine:
   environment:
-    TESTCMD: 'python -V && pip install .[tests] && flake8 && nosetests -vv --exclude=compare_implementations_test'
+    TESTCMD: 'python -V && pip install .[tests] && flake8 && nosetests --with-coverage --cover-package=s2sphere -vv --exclude=compare_implementations_test'
```
Comment (svenkreiss) [inline circle.yml]: Use `source` or `include` in a `setup.cfg` with a `[coverage]` section instead.

File: circle.yml [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
 
 test:
   override:
-    - deactivate && pyenv shell 2.7.11 && bash -c "$TESTCMD"
-    - deactivate && pyenv shell 3.4.4 && bash -c "$TESTCMD"
-    - deactivate && pyenv shell 3.5.1 && bash -c "$TESTCMD"
+    - deactivate && pyenv shell 2.7.11 && bash -c "$TESTCMD" && coveralls
+    - deactivate && pyenv shell 3.4.4 && bash -c "$TESTCMD" && coveralls
+    - deactivate && pyenv shell 3.5.1 && bash -c "$TESTCMD" && coveralls
```
Comment (svenkreiss) [inline circle.yml]: run `coveralls` after the tests

**Comment by danvk** [comment]
@svenkreiss PTAL!

File: circle.yml (hunk 1 above)
Comment (danvk) [inline circle.yml]: Done.

**Comment by danvk** [comment]
Note that the `.coverage` files from individual runs are _not_ summed: you can see this because the `import` statements are only run once, not 3x.

**Comment by svenkreiss** [comment]
LGTM

**Pull request merged**